R/maic_anchored.R

```diff
@@ -293,13 +347,188 @@ maic_anchored_tte <- function(res,
+  if (!is.null(weights_object$boot)) {
+    keep_rows <- setdiff(seq_len(nrow(weights_object$data)), weights_object$rows_with_missing)
+    boot_ipd_id <- weights_object$data[keep_rows, "USUBJID", drop = FALSE]
+
+    boot_ipd <- merge(boot_ipd_id, ipd, by = "USUBJID", all.x = TRUE)
+    if (nrow(boot_ipd) != nrow(boot_ipd_id)) stop("ipd has multiple observations for some patients")
+    boot_ipd <- boot_ipd[match(boot_ipd$USUBJID, boot_ipd_id$USUBJID), ]
+
+    stat_fun <- function(data, index, w_obj, eff_measure) {
+      r <- dynGet("r", ifnotfound = NA) # Get bootstrap iteration
+      if (!is.na(r)) {
+        if (!all(index == w_obj$boot[, 1, r])) stop("Bootstrap and weight indices don't match")
+        boot_ipd <- data[w_obj$boot[, 1, r], ]
+        boot_ipd$weights <- w_obj$boot[, 2, r]
+      }
+
+      boot_binobj_dat_adj <- glm(RESPONSE ~ ARM, boot_ipd, weights = boot_ipd$weights, family = glm_link)
+      boot_AC_est <- coef(boot_binobj_dat_adj)[2]
+      boot_AC_var <- vcov(boot_binobj_dat_adj)[2, 2]
+
+      boot_res_AC <- list(est = boot_AC_est, se = sqrt(boot_AC_var))
+      boot_res_AB <- bucher(boot_res_AC, res_BC, conf_lv = 0.95)
+
+      c(
+        est_AB = boot_res_AB$est,
+        var_AB = boot_res_AB$se^2,
+        se_AB = boot_res_AB$se,
+        est_AC = boot_res_AC$est,
+        se_AC = boot_res_AC$se,
+        var_AC = boot_res_AC$se^2
+      )
+    }
+
+    # Revert seed to how it was for weight bootstrap sampling
+    old_seed <- globalenv()$.Random.seed
+    on.exit(suspendInterrupts(set_random_seed(old_seed)))
+    set_random_seed(weights_object$boot_seed)
+
+    R <- dim(weights_object$boot)[3]
+    boot_res <- boot(
+      boot_ipd,
+      stat_fun,
+      R = R,
+      w_obj = weights_object,
+      eff_measure = eff_measure,
+      strata = weights_object$boot_strata
+    )
+    boot_ci <- boot.ci(boot_res, type = boot_ci_type, w_obj = weights_object)
+
+    l_u_index <- switch(boot_ci_type,
+      "norm" = list(2, 3, "normal"),
+      "basic" = list(4, 5, "basic"),
+      "stud" = list(4, 5, "student"),
+      "perc" = list(4, 5, "percent"),
+      "bca" = list(4, 5, "bca")
+    )
+
+    transform_estimate <- switch(eff_measure,
+      "RD" = function(x) x * 100,
+      "RR" = exp,
+      "OR" = exp
+    )
+
+    res$inferential[["boot_est"]] <- boot_res
+    boot_res_AB <- list(
+      est = transform_estimate(boot_res$t0[1]),
+      se = NA,
+      ci_l = transform_estimate(boot_ci[[l_u_index[[3]]]][l_u_index[[1]]]),
+      ci_u = transform_estimate(boot_ci[[l_u_index[[3]]]][l_u_index[[2]]]),
+      pval = NA
+    )
+  } else {
+    res$inferential[["boot_est"]] <- NULL
+  }
+
+  # : make analysis report table
+  tags <- paste0(c("IPD/", "weighted IPD/", "AgD/"), endpoint_name)
+  res$inferential[["report_overall_robustCI"]] <- rbind(
+    report_table_binary(binobj_ipd, tag = tags[1], eff_measure = eff_measure),
+    report_table_binary(binobj_ipd_adj, res_AB, tag = tags[2], eff_measure = eff_measure),
```

Comment:
@hoppanda Trying to follow what's being reported here. I think `res_AB` should actually be the results for the weighted `AC` comparison, right? The A v B should only appear in the last line?
eg
A v C
A v C weighted
B v C
A v B
```suggestion
    report_table_binary(binobj_ipd_adj, res_AB, tag = tags[2], eff_measure = eff_measure),
```

Reply:
I've updated this. Can you check @hoppanda ?